performance-test/src/puppeteer-test.ts

```diff
@@ -0,0 +1,362 @@
+require('dotenv').config({ path: 'src/.env' })
+import puppeteer from 'puppeteer'
+import { v4 } from 'uuid'
+const fs = require('fs')
+const path = require('path')
+const AWS = require('aws-sdk')
+const moveFile = require('move-file')
+
+const BRANCH_NAME = process.env.BRANCH_NAME
+const PROJECT_ID = '5596ecdd'
+// const EDITOR_URL = `http://localhost:8000/p/39c427a7-hypnotic-king/` //locally
+const EDITOR_URL = `https://utopia.pizza/project/${PROJECT_ID}/?branch_name=${BRANCH_NAME}` //server, whenever push to server make sure this line is active
+
+// this is the same as utils.ts@defer
+function defer() {
+  var res, rej
+  var promise = new Promise((resolve, reject) => {
+    res = resolve
+    rej = reject
+  })
+  Object.defineProperty(promise, 'resolve', { value: res })
+  Object.defineProperty(promise, 'reject', { value: rej })
+
+  return promise
+}
+
+function consoleDoneMessage(page: puppeteer.Page) {
+  return new Promise<void>((resolve, reject) => {
+    page.on('console', (message) => {
+      if (message.text().includes('SCROLL_TEST_FINISHED')) {
+        // the editor will console.info('SCROLL_TEST_FINISHED') when the scrolling test is complete.
+        // we wait until we see this console log and then we resolve the Promise
+        resolve()
+      }
+    })
+  })
+}
+
+export const testScrollingPerformance = async function () {
+  const browser = await puppeteer.launch({
+    args: ['--no-sandbox', '--enable-thread-instruction-count'],
+    headless: false,
```

Comment:
it seems like you accidentally committed `headless: false` here, which breaks the script on the server.

to avoid this mistake in the future, we should probably turn this into an environment variable (similar to how `process.env.BRANCH_NAME` works) so you don't need to change it for debugging, because it is always a risk that someone will forget to change it back.